### ISSUE TITLE
Ability to create a patient when adding a related person that does not exist

### ISSUE THREAD

**Issue by jackcmeyer** [HospitalRun/hospitalrun-frontend#2106]
## 🚀 Feature Proposal
When creating a Related Person, it assumes that the patient that you want to add as a related person already exists. If the related person does not exist, it is cumbersome to exit the related person modal and go create that patient, then come back to finally add the related person. 

Users would like the ability to have some sort of "quick add" that allows me to create a new patient without leaving the related person modal.

## Motivation
Be able to quickly add related persons that aren't currently existing patients.

**Comment by WaseemAlkurdi** [comment]
Example scenario: Let's take the context of a fertility/family planning clinic, where the doctor is creating an entry for an example patient John Doe:
![image](https://user-images.githubusercontent.com/15712695/83309074-00d00d00-a211-11ea-9cb1-7ac322341a7a.png)

That John Doe is married to Jane Doe, _who should share the same record, as the clinic's context  needs that_ . Currently, we have to create a separate entry for her:
![image](https://user-images.githubusercontent.com/15712695/83309141-2b21ca80-a211-11ea-96fc-a8b35387f036.png)
then we would have to search for John Doe's record, open it, select Related Persons from there, Add Related Person, then search Jane Doe and select her as Wife. (The "Patients List" page didn't work at the time of posting this, but I will update with screenshots once it becomes available again).

In addition to being cumbersome and very anti-HospitalRun, this approach simply won't work in a fertility clinic, where a couple, not an individual patient, is the focus of the diagnosis and treatment.
Currently, the changes to one person's record would not automatically propagate to the other person's, as the database would consider them as separate patients who just happen to be related. But, as we have seen, this might not be wanted in an obstetrics/gynaecology/fertility/family practice, among others.

A solution would be to move Related Persons to the Add Patient page, where the doctor would be able to enter both John Doe and Jane Doe's details at once, then it would offer the user a checkbox to "join" records (like in the case of a fertility clinic), or leave them detached.

**Comment by JDarke** [comment]
I'd like to take a shot at this one, please.

**Comment by fox1t** [comment]
@JDarke nice! We need to wait for this since we need to define more specific requirements. Would you like to be involved also in requirements planning and user story definition?

**Comment by JDarke** [comment]
Absolutely!

I've got my own projects in react/typescript to finish off today, then I'll take a deeper look at the issue and come back with some suggestions.

**Comment by kumikokashii** [comment]
> A solution would be to move Related Persons to the Add Patient page, where the doctor would be able to enter both John Doe and Jane Doe's details at once, then it would offer the user a checkbox to "join" records (like in the case of a fertility clinic), or leave them detached.

Please note that this "join" feature will be a separate ticket if wanted.

**Comment by JDarke** [comment]
So it'd been a while since I last updated my hr-fe fork, and having just done so and pulled to my local repo, I'm getting a failure to compile.

Is anyone else getting this error?

`TypeScript error in /.../hospitalrun-frontend/src/components/Navbar.tsx(70,11)`

**Comment by tehKapa** [comment]
No. Try deleting node_modules folder, yarn.lock, package-lock.json and relaunch an `npm i`

<img width="657" alt="Screenshot 2020-06-09 at 20 28 45" src="https://user-images.githubusercontent.com/6388707/84185799-d6434700-aa8f-11ea-987a-44a598d9def5.png">

**Comment by fox1t** [comment]
> That John Doe is married to Jane Doe, _who should share the same record, as the clinic's context needs that_ . Currently, we have to create a separate entry for her:
> In addition to being cumbersome and very anti-HospitalRun, this approach simply won't work in a fertility clinic, where a couple, not an individual patient, is the focus of the diagnosis and treatment.
> Currently, the changes to one person's record would not automatically propagate to the other person's, as the database would consider them as separate patients who just happen to be related. But, as we have seen, this might not be wanted in an obstetrics/gynaecology/fertility/family practice, among others.
> 

Hi, I thought a lot about what you said here, @WaseemAlkurdi, and I disagree that "auto-propagation" and "sharing" would be what we want here. There are more contexts where sharing diagnosis and conditions is more harmful than useful. However I perfectly understand your point of being able to see the relations between patients and fastly check conditions of relative ones. We are going to make this more explicit and easy to do.

For sharing the record between patient thing, I think we can make an optional flag that, that renders a clickable link on a patient detail that let the user (doctor) go directly to an "important external patient condition". However we need to put some boundaries on what can be considered "a linked condition". We will work on this as soon as possible and we will publish a user story about this.

**Comment by fox1t** [comment]
@JDarke What you propose is really nice. I like it a lot. @HospitalRun/core-contributor what do you think about it? If we all agree we can elaborate a bit and make a new issue/PR to handle it.

**Comment by JDarke** [comment]
Thanks.

I also noticed a potential issue:  when searching for a match in existing patients, it only searches by name, and doesn't display any other info to the user before they confirm the match as a related person.

This seems likely to cause issues if say, I'm searching for a person with a fairly common name, and they are presented as a match when in fact I'm looking for someone completely different who is not yet on the system.

![hrfe2](https://user-images.githubusercontent.com/52716187/84676561-91069580-af25-11ea-8d38-3c72ef7048c5.jpg)

It seems as though it should also display part of their address, for example, so that the user can distinguish between multiple matches with the same name, or can tell straightaway if the match is someone in a different city, when it's a person from the same household that they're looking for.

I'm not sure of the etiqutte/process here for reporting issues, as I can't join the slack channel without a hospitalrun account, so I didn't want to raise a ticket for it without asking first...

**Comment by WaseemAlkurdi** [comment]
> > That John Doe is married to Jane Doe, _who should share the same record, as the clinic's context needs that_ . Currently, we have to create a separate entry for her:
> > In addition to being cumbersome and very anti-HospitalRun, this approach simply won't work in a fertility clinic, where a couple, not an individual patient, is the focus of the diagnosis and treatment.
> > Currently, the changes to one person's record would not automatically propagate to the other person's, as the database would consider them as separate patients who just happen to be related. But, as we have seen, this might not be wanted in an obstetrics/gynaecology/fertility/family practice, among others.
> 
> Hi, I thought a lot about what you said here, @WaseemAlkurdi, and I disagree that "auto-propagation" and "sharing" would be what we want here. There are more contexts where sharing diagnosis and conditions is more harmful than useful. However I perfectly understand your point of being able to see the relations between patients and fastly check conditions of relative ones. We are going to make this more explicit and easy to do.
> 
> For sharing the record between patient thing, I think we can make an optional flag that, that renders a clickable link on a patient detail that let the user (doctor) go directly to an "important external patient condition". However we need to put some boundaries on what can be considered "a linked condition". We will work on this as soon as possible and we will publish a user story about this.

Thanks for the insight!
As for the "auto-propagation" part, you definitely have a point here. But the rest of the objection isn't as clear to me, especially when our problem (sans record linking, which is acceptable for the reasons you cite) was very eloquently solved by @JDarke in the screenshots provided above. Can you tell me what I might be missing?

**Comment by WaseemAlkurdi** [comment]
> Thanks.
> 
> I also noticed a potential issue: when searching for a match in existing patients, it only searches by name, and doesn't display any other info to the user before they confirm the match as a related person.
> 
> This seems likely to cause issues if say, I'm searching for a person with a fairly common name, and they are presented as a match when in fact I'm looking for someone completely different who is not yet on the system.
> 
> ![hrfe2](https://user-images.githubusercontent.com/52716187/84676561-91069580-af25-11ea-8d38-3c72ef7048c5.jpg)
> 

Nicely spotted! 

> I'm not sure of the etiquette/process here for reporting issues, as I can't join the slack channel without a hospitalrun account, so I didn't want to raise a ticket for it without asking first...

You could actually join by creating a Slack account (no HospitalRun email address needed), I tried that and it worked. You can request an invite using this link: https://hospitalrun-slackin.herokuapp.com/

**Comment by lauggh** [comment]
Would it be possible to also pull in a date of birth next to the person's name as a qualifier? This would help identify patients as well as show age. 

`Jane Doe (1980-06-17)`

**Comment by JDarke** [comment]
Yup, seems sensible.

It's been raised as a separate issue - #2132  .

**Comment by JDarke** [comment]
Just realised I haven't actually been assigned to this yet...can someone assign me, please?  

I'll have a WIP ready for review shortly.
